### PR TITLE
Feat(Enhancement): enhance validation for deps, mount and file paths

### PR DESCRIPTION
**Pull request by Callum-HQ**

### CODE REVIEW

File: DiskUtils.sh [hunk 1]
```diff
@@ -162,18 +163,18 @@ format_device() {
...
         ;;
       3)
-        mkfs.fat -F32 "$device"
-        log "${GREEN}Device formatted as FAT32 filesystem successfully.${NC}"
+        mkfs.fat -F32 "$disk"
+        log "${GREEN}Disk formatted as FAT32 filesystem successfully.${NC}"
         ;;
       4)
-        mkfs.exfat "$device" || log "${RED}Failed to format device as exFAT. Ensure exfat-utils is installed.${NC}"
+        mkfs.exfat "$disk" || log "${RED}Failed to format disk as exFAT. Ensure exfat-utils is installed.${NC}"
```
Comment (Kairos-T) [inline DiskUtils.sh]: exfat-utils are deprecated. `exfatprogs` are used now

Comment (Callum-HQ) [reply DiskUtils.sh]: tahnk

Comment (Kairos-T) [reply DiskUtils.sh]: sorry my comment was a little misleading. you should check using `mkfs.exfat` instead and ensure the output doesn't say like  `Command 'mkfs.exfat' not found, but can be installed with:`

**Comment by Callum-HQ** [comment]
thanks

**Pull request merged**